packages/vite/src/node/plugins/resolve.ts

```diff
@@ -152,6 +153,29 @@ export function resolvePlugin(resolveOptions: InternalResolveOptions): Plugin {
         scan: resolveOpts?.scan ?? resolveOptions.scan,
       }
 
+      const resolveSubpathImports = (id: string, importer?: string) => {
+        if (!importer || !id.startsWith(subpathImportsPrefix)) return
+        const basedir = path.dirname(importer)
+        const pkgJsonPath = lookupFile(basedir, ['package.json'], {
+          pathOnly: true,
+        })
```

Comment:
Note to self. `exports` only work for root `package.json`, but node decided that `imports` doesn't need to and works for the closest `package.json`, so this is fine. [Stackblitz](https://stackblitz.com/edit/node-tka6oy?file=nested%2Fpackage.json,nested%2Findex.js)